### ISSUE TITLE
Run notebook locally without requesting token each time

### ISSUE THREAD

**Issue by DaveVoyles** [jupyter/notebook#3607]
I'm not sure why it requests a token or password each time. It would be fine if it worked, but it is oddly inconsistent. 

**OS:** Windows 10 

In terminal: ```jupyter notebook```

Returns:

```bash
[I 18:01:41.360 NotebookApp] JupyterLab beta preview extension loaded from C:\ProgramData\Anaconda3\lib\site-packages\jupyterlab
[I 18:01:41.360 NotebookApp] JupyterLab application directory is C:\ProgramData\Anaconda3\share\jupyter\lab
[I 18:01:42.467 NotebookApp] Serving notebooks from local directory: D:\OneDrive - Microsoft\Formerly shared with Everyone\Documents\Hacks-Events\OpenHack
[I 18:01:42.468 NotebookApp] 0 active kernels
[I 18:01:42.468 NotebookApp] The Jupyter Notebook is running at:
[I 18:01:42.468 NotebookApp] http://localhost:8888/
[I 18:01:42.468 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[I 18:01:42.720 NotebookApp] 302 GET /tree (::1) 1.00ms
[I 18:01:51.220 NotebookApp] 302 POST /login?next=%2Ftree (::1) 1.00ms
[W 18:01:59.341 NotebookApp] 404 GET /nbextensions/canopy/training/exercise.js?v=20180511180140 (::1) 7.00ms referer=http://localhost:8888/notebooks/Challenge-3-Introduction-to-custom-ml.ipynb
[I 18:02:00.269 NotebookApp] Kernel started: e0aacc0d-b684-4f7a-a3f8-a7974be03848
[W 18:02:00.563 NotebookApp] 404 GET /nbextensions/widgets/notebook/js/extension.js?v=20180511180140 (::1) 2.00ms referer=http://localhost:8888/notebooks/Challenge-3-Introduction-to-custom-ml.ipynb
[W 18:02:01.959 NotebookApp] Forbidden
```

And from then on, I can't save anything. I try to create a password, it accepts it, but then when I try to save, it reads "Forbidden".

I've tried clearing my browsers completely, restarting my machine, and using Firefox, Edge, and Chrome.

**Comment by DaveVoyles** [comment]
I just learned that you can disable security, ere: [](https://jupyter-notebook.readthedocs.io/en/stable/security.html)


Alternatives to token authentication
If a generated token doesn’t work well for you, you can set a password for your notebook. jupyter notebook password will prompt you for a password, and store the hashed password in your jupyter_notebook_config.json.

New in version 5.0: jupyter notebook password command is added.

It is possible disable authentication altogether by setting the token and password to empty strings, but this is NOT RECOMMENDED, unless authentication or access restrictions are handled at a different layer in your web application:

c.NotebookApp.token = ''
c.NotebookApp.password = ''

Now all of my saving issues are gone, everything works.

**Comment by LegitStack** [comment]
I'd like to comment that there is a case where you don't want to have a token, even if it means your system is vulnerable.

We encapsulate all our data science projects in docker containers. These containers are removed as soon as we exit the container, thus system vulnerability is not a concern, no data is saved on the container themselves. Most of the data exploration we do is in jupyter noteoboks, thus the container serves as an encapsulated configured environment that only runs jupyter notebook, and does so automatically when you start the container.

Until I found this solution we had to copy and paste the token from the container repl into a browser every time. so thanks for this solution!
